cdap-metadata-spi/src/main/java/io/cdap/cdap/spi/metadata/ProgramRun.java

```diff
@@ -0,0 +1,300 @@
+/*
+ * Copyright © 2022 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.spi.metadata;
+
+import io.cdap.cdap.api.annotation.Beta;
+import io.cdap.cdap.api.plugin.PluginClass;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * Represents a program run for which lineage is generated.
+ */
+@Beta
+public class ProgramRun {
+
+  private final String id;
+  private final String programId;
+  private final String namespace;
+  private final String application;
+  private final Long startTimeMs;
+  private final Long endTimeMs;
+  private final String status;
+  private final Map<String, String> properties;
+
+  private ProgramRun(String id, String programId, String namespace, String application, Long startTimeMs, Long endTimeMs,
+                     String status, Map<String, String> properties) {
+    this.id = id;
+    this.programId = programId;
+    this.namespace = namespace;
+    this.application = application;
+    this.startTimeMs = startTimeMs;
+    this.endTimeMs = endTimeMs;
+    this.status = status;
+    this.properties = properties;
+  }
+
+  /**
+   * Validates the {@link PluginClass}
+   *
+   * @throws IllegalArgumentException if any of the required fields are invalid
+   */
+  public void validate() {
```

Comment:
`validate()` shouldn't be an instance method. You either validate in the constructor or validate through builder.